src/cryptoadvance/specter/templates/device/new_device.jinja

```diff
@@ -7,77 +7,185 @@
+						<img src="/static/img/usb_tiny.svg" height="12px">Get over USB
+					</button>
 				</div>
-			</label>
-		</div>
-		<div class="row">
-			<button type="button" class="btn" style="margin: 2px;" id="askhwi">
-				<img src="/static/img/usb_tiny.svg" height="12px">Get over USB
-			</button>
-		</div>
-		<br>
-		<div class="note">
-			One line per xpub. Ideally with derivation path in the form "<b>[fingerprint/derivation/path]xpub</b>".<br>
-			Examples:<br>
+				<br>
+				<div class="note">
+					One line per xpub. Ideally with derivation path in the form "<b>[fingerprint/derivation/path]xpub</b>".<br>
+					Examples:<br>
 			<pre style="font-size: 1.2em">
 [f79ec910/84'/1'/0']tpubDA5h1hw24fzoBi9...pR1DiM4EBu
 [F79EC910/48h/1h/0h/2h]Vpub5DRk6RvRAjZf9rmZV1Q8...TZSDQL67e5i
 upub5En4f7k8gaG2KDHvBeEYox...rFpJRHpiZ4DE
 			</pre>
-			Loading from file supports ColdCard and Electrum wallet format.<br>
-		</div>
-		<button type="submit" class="btn centered" name="action" value="morekeys">Continue</button>
-	</form>
+					Loading from file supports ColdCard and Electrum wallet format.<br>
+				</div>
+				<button type="submit" class="btn centered" name="action" value="newcolddevice">Continue</button>
+			</div>
+			<div id="hot_device" class="hidden">
+				<p>Enter you mnemonic here or use the randomly generated one:</p>
+				<p><b>Make sure to backup these words!</b></p>
+				{% set wordslist = mnemonic.split(' ') %}
+				<nav class="row">
+					<button type="button" id="generate_mnemonic_btn" class="btn radio left checked" onclick="setMnemonicView('generate')"> Generate </button>
+					<button type="button" id="import_mnemonic_btn" class="btn radio right" onclick="setMnemonicView('import')"> Import </button>
+				</nav>
+				<input id="mnemonic_value" type="hidden" name="mnemonic" value="{{ mnemonic }}"/>
+				<textarea id="import_mnemonic" class="hidden" id="txt" placeholder="Enter your mnemonic here"></textarea>
+				<div id="generated_mnemonic">
+					<div class="row center" style="max-height: 30px;">
+						<fieldset style="border:none; display: inline;"> 
+							<select name="strength" id="strength">
+								<option value="128">12 words</option>
+								<option value="256">24 words</option>
+							</select>
+						</fieldset>
+						<button type="submit" class="btn" name="action" value="generatemnemonic" style="max-width: 200px;">Generate New Mnemonic</button>
+					</div>
+					<br>
+					<table>
+						{% for i in range(wordslist|length // 4) %}
+							<tr>
+								<td>{{ 4 * i + 1 }}. {{ wordslist[4 * i] }}.</td>
+								<td>{{ 4 * i + 2 }}. {{ wordslist[4 * i + 1] }}</td>
+								<td>{{ 4 * i + 3 }}. {{ wordslist[4 * i + 2] }}</td>
+								<td>{{ 4 * i + 4 }}. {{ wordslist[4 * i + 3] }}</td>
+							</tr>
+						{% endfor %}
+					</table>
+				</div>
+				<br>
+				<label>Passphrase: </label>
+				<input type="password" name="passphrase" class="inline" placeholder="Wallet passphrase">
```

Comment:
To avoid confusion we should call it "Encryption password" as it's not a bip39 password (unlike passwords in hardware wallet).